nemo_skills/evaluation/metrics/base.py

```diff
@@ -29,6 +30,92 @@ def get_metrics(self):
     def reset(self):
         pass
 
+    @abc.abstractmethod
+    def get_prediction_results(self, prediction):
+        pass
+
+    def get_majority_at_k(
+        self, agg_mode_dict, predicted_answers, pred_keys=None, predictions=None, prediction_results=None
```

Comment:
do we need prediction_results parameter? I see it only used in math metrics but seems that the provided value is the same as what is computed in this function anyway?

Reply:
No we don't. I made this a generic function. But now I'll move it inside the math metric file so will remove the unecessary parameters.